class.c

```diff
@@ -175,6 +175,8 @@ XS(injected_constructor)
 
     SV *self = sv_2mortal(newRV_noinc(instance));
 
+    PUSHSTACKi(PERLSI_CONSTRUCTOR);
```

Comment:
Any particular reason for not using `push_stackinfo` here?

Reply:
Yes, for SP